File: docs/WidgetMill/packages/spec/src/nodes/fractionallySizedBox.ts
import { z } from "zod";
import { Alignment } from "../complex/alignment";
import { NodeRef } from "../node-ref";

/**
 * FractionallySizedBox — dimensiona o filho como fração do espaço do **pai**
 * (complementa o `Dimension` de width/height, que é relativo à **tela**).
 */
export const FractionallySizedBoxNode = z.object({
  type: z.literal("fractionallySizedBox"),
  props: z
    .object({
      widthFactor: z.number().optional(),
      heightFactor: z.number().optional(),
      alignment: Alignment.optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/fractionally_sized_box.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// FractionallySizedBox — dimensiona o filho como fração do espaço do pai.
Widget buildFractionallySizedBox(
    BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return FractionallySizedBox(
    widthFactor: parseDouble(p['widthFactor']),
    heightFactor: parseDouble(p['heightFactor']),
    alignment: alignmentFrom(p['alignment']) ?? Alignment.center,
    child: r.maybeRender(context, node.child),
  );
}
